tests/loops/optimization/test_optimizer_loop.py

```diff
@@ -109,16 +109,13 @@ class CurrentModel(BoringModel):
         def training_step(self, batch, batch_idx, optimizer_idx):
             return super().training_step(batch, batch_idx)
 
-        def training_epoch_end(self, outputs):
-            assert len(outputs[0]) == sum(idx == 0 for idx, _ in expected)
-            assert len(outputs[1]) == sum(idx == 1 for idx, _ in expected)
```

Comment:
We were using the `training_epoch_end` outputs format as a side effect to check the optimizer indices called, this is not possible anymore and can be considered a breaking change.

The expected format is still checked after `fit` below

Reply:
note, if one needs the index of the optimizer, then that can be put as an output in the training step and it will show up here.